casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -49,7 +49,6 @@ trait RuntimeManager[F[_]] {
   ): F[(StateHash, StateHash, Seq[InternalProcessedDeploy])]
   def storageRepr(hash: StateHash): F[Option[String]]
   def computeBonds(hash: StateHash): F[Seq[Bond]]
-  def computeBalance(start: StateHash)(user: ByteString): F[Long]
```

Comment:
When would the balance check take place? Before execution? Notice that the deploy can also access the deployers vault and spend it to a place where there's nothing left. So there needs to be a transfer before the deploy being executed, and then a transfer of the leftovers back to the deployer's vault.

Reply:
> Notice that the deploy can also access the deployers vault and spend it to a place where there's nothing left.

Would you explain this a bit more? The balance check would occur before the deploy is executed.

Reply:
Nvm. I see.

Reply:
Say Bob deploys with phloLimit 100 and phloRate 1. His balance is 200. His deploy does a transfer of 150 rev from his vault to Alice. His deploy costs 75 phlo (= 75 rev)

With two transfers, Bob is going to hit a 'insufficient funds' before attempting the transfer (only 125 is left at deploy start). No further processing is needed after an OOPE. We need to call a secured method from PoS contract to do the refund, but we need to figure that out anyway for other casperwork.

With balance check + transfer after deploy, Bob's deploy won't be seen by the network at all - but the original validator will need to process its entirety. When OOPE happens, additional rholang execution is needed.
When the deploy finishes, we need to do another balance check, 
reset the deploy's state change if not enough is left to cover execution costs, and transfer the cost of phlo used. Otherwise the example discussed above ends up with an 'unexpected error during payment deploy'.

Reply:
would it make sense to make this discussion part of a less volatile piece of data? (wiki page?)